### PR TITLE
feat(auth-plugin): plugins server mode

### PR DESCRIPTION
**Pull request by olisajc**
Delegate server-side token acquisition entirely to auth plugins via a new ActivateServerMode RPC. Plugins now own provider-specific flows (client_credentials, WIF, OBO) internally -- scafctl passes opaque JSON settings at startup and calls the standard Handler interface

- Add ActivateServerMode to plugin gRPC client/server and wrappers
- Add disabledHandler + Registry.Disable() to block CLI-only handlers in API mode (prevents device-code flows in non-interactive server)
- Serve command activates configured plugins then disables the rest
- Catalog/HTTP provider call handler.GetToken()/InjectAuth() directly
- Hostname token resolution uses auth.GetHandler() instead of tokenprovider
- calling in different contexts delegate for downstream api calls pkg/provider/builtin/httpprovider/http.go and server when using server context like in BridgeAuthToRegistry
- serverIdentity and token provider packages are removed .

BREAKING CHANGE:
- Remove pkg/serveridentity (entra, github, registry)
- Remove pkg/tokenprovider (registry, adapter, build, callerscope)
- Remove pkg/config IdentityProvider struct + validation.go
- functions and types like enumeratorConfig & BridgeAuthToRegistry signature takes auth.Handler instead of string like it was at commit d3c2593
- Remove WithServerTokenProviderRegistry API server option
- APIServerAuth.Handlers is now map[string]any (opaque plugin config)

## Description

Brief description of the changes.

## Related Issues

Closes #

## Type of Change

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Documentation update
- [ ] Refactoring

## Checklist

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) guide
- [x] My commits follow conventional commit format
- [x] I have added/updated tests for my changes
- [x] `go test ./...` passes
- [x] `task lint` passes
- [x] I have signed off my commits (`git commit -s -S`)

### CODE REVIEW

File: pkg/catalog/auth_bridge.go [hunk 1]
```diff
@@ -33,79 +30,44 @@ type RegistryUsernameProvider interface {
...
-	authRegistry := auth.RegistryFromContext(ctx)
-	if authRegistry == nil {
-		return "", fmt.Errorf("no auth registry in context for provider %q", provider)
-	}
-	handler, err := authRegistry.Get(provider)
+	token, err := handler.GetToken(ctx, opts)
 	if err != nil {
-		return "", fmt.Errorf("auth provider %q not found in registry: %w", provider, err)
+		return "", "", fmt.Errorf("get token from %s handler: %w", handler.Name(), err)
 	}
```
[the remark this answers is not in the document]
Comment (olisajc) [reply pkg/catalog/auth_bridge.go:52]: Fixed. Added nil handler guard and empty token check in `BridgeAuthToRegistry`. Now returns a descriptive error for both cases instead of panicking.

File: pkg/catalog/auth_bridge_test.go [hunk 2]
```diff
@@ -149,16 +128,17 @@ func TestBridgeAuthToRegistry_TokenError(t *testing.T) {
...
-func TestBridgeAuthToRegistry_NoRegistry(t *testing.T) {
-	// No tokenprovider registry in context at all.
-	_, _, err := BridgeAuthToRegistry(context.Background(), "github", "ghcr.io", "")
+func TestBridgeAuthToRegistry_NoHandler(t *testing.T) {
+	// Nil handler should error.
+	mock := auth.NewMockHandler("missing")
+	mock.GetTokenErr = fmt.Errorf("handler not configured")
+	_, _, err := BridgeAuthToRegistry(context.Background(), mock, "ghcr.io", "")
 	require.Error(t, err)
 }
```
[the remark this answers is not in the document]
Comment (olisajc) [reply pkg/catalog/auth_bridge_test.go:150]: Fixed. The test now actually passes `nil` as the handler and verifies the error message. Also added a separate `TestBridgeAuthToRegistry_EmptyToken` test for the empty-token path.

File: pkg/auth/registry.go [hunk 3]
```diff
@@ -104,6 +104,22 @@ func (r *Registry) Unregister(name string) error {
...
+	defer r.mu.Unlock()
+
+	handler, exists := r.handlers[name]
+	if !exists {
+		return fmt.Errorf("%w: %s", ErrHandlerNotFound, name)
+	}
+
+	r.handlers[name] = newDisabledHandler(handler, reason)
+	return nil
+}
```
[the remark this answers is not in the document]
Comment (olisajc) [reply pkg/auth/registry.go:127]: Fixed. `Disable` now detects if the handler is already a `*disabledHandler` and updates the reason in-place instead of double-wrapping.

File: pkg/solution/prepare/prepare.go [hunk 4]
```diff
@@ -572,28 +572,41 @@ func NewDefaultGetter(ctx context.Context, noCache bool) get.Interface {
...
+
+				if handlerName == "" {
+					return nil
+				}
+				handler, err := auth.GetHandler(ctx, handlerName)
+				if err != nil {
+					return nil
 				}
-				return catalog.InferAuthHandler(registry, customHandlers)
+				return handler
```
[the remark this answers is not in the document]
Comment (olisajc) [reply pkg/solution/prepare/prepare.go:610]: Fixed. Added verbose-level logging (`lgr.V(1).Info`) when `auth.GetHandler` fails, including the handler name, registry, and error. Still returns nil to allow anonymous fallback, but the failure is now observable in debug logs.

File: pkg/cmd/scafctl/serve/serve.go [hunk 5]
```diff
@@ -477,3 +464,88 @@ func buildPluginPool(ctx context.Context, cfg *config.Config, fetcher *plugin.Fe
...
+
+		settings, err := authPluginSettings(cfg, handlerName)
+		if err != nil {
+			return fmt.Errorf("auth handler %q settings: %w", handlerName, err)
+		}
+		if settings == nil {
+			return fmt.Errorf("auth handler %q has no server mode settings configured", handlerName)
+		}
+
+		if err := activator.ActivateServerMode(ctx, settings); err != nil {
```
[the remark this answers is not in the document]
Comment (olisajc) [reply pkg/cmd/scafctl/serve/serve.go:512]: This is not a type mismatch. Per the [Go spec assignability rules](https://go.dev/ref/spec#Assignability): a value `x` of type `V` is assignable to a variable of type `T` when `V` and `T` have identical underlying types. Since `json.RawMessage` is defined as `type RawMessage []byte`, `[]byte` and `json.RawMessage` share the same underlying type (`[]byte`) and are freely assignable in either direction without conversion. All CI checks (including `go build` and `go vet`) pass cleanly.

File: pkg/plugin/wrapper_auth.go [hunk 6]
```diff
@@ -590,3 +593,7 @@ func KillAllAuthHandlers(clients []*AuthHandlerClient) {
 		}
 	}
 }
+
+func (w *AuthHandlerWrapper) ActivateServerMode(ctx context.Context, settings json.RawMessage) error {
+	return w.client.ActivateServerMode(ctx, settings)
+}
```
[the remark this answers is not in the document]
Comment (olisajc) [reply pkg/plugin/wrapper_auth.go:599]: Not a type mismatch. `AuthHandlerWrapper.ActivateServerMode` receives `json.RawMessage` and passes it to `AuthHandlerClient.ActivateServerMode` which takes `[]byte`. Per [Go assignability rules](https://go.dev/ref/spec#Assignability), `json.RawMessage` (underlying type `[]byte`) is directly assignable to `[]byte` — no conversion needed. CI confirms this compiles and passes.

File: pkg/plugin/client.go [hunk 7]
```diff
@@ -581,6 +581,13 @@ func (c *AuthHandlerClient) HostServiceID() uint32 {
 	return 0
 }
 
+func (c *AuthHandlerClient) ActivateServerMode(ctx context.Context, settings []byte) error {
+	if sm, ok := c.plugin.(*AuthHandlerGRPCClient); ok {
+		return sm.ActivateServerMode(ctx, settings)
+	}
+	return fmt.Errorf("plugin does not support server mode")
```
[the remark this answers is not in the document]
Comment (olisajc) [reply pkg/plugin/client.go:588]: Same as the other two — not a type mismatch. `[]byte` passed to a `json.RawMessage` parameter is valid per [Go assignability rules](https://go.dev/ref/spec#Assignability) since they share the same underlying type. No conversion or import needed. CI green.

File: pkg/provider/builtin/httpprovider/http.go [hunk 8]
```diff
@@ -627,6 +603,50 @@ func (p *HTTPProvider) Execute(ctx context.Context, input any) (*provider.Output
...
+		refreshOpts := tokenOpts
+		refreshOpts.ForceRefresh = true
+		t, refreshErr := handler.GetToken(unauthCtx, refreshOpts)
+		if refreshErr != nil {
+			return "", refreshErr
+		}
+		return fmt.Sprintf("%s %s", t.TokenType, t.AccessToken), nil
+	}
+
+	return token, nil
```
[the remark this answers is not in the document]
Comment (olisajc) [reply pkg/provider/builtin/httpprovider/http.go:653]: Good catch. Added defensive nil/empty-token checks for both the initial `GetToken` and the 401 refresh callback. Now returns a descriptive error instead of panicking or sending an invalid Authorization header.

**Review by kcloutie — APPROVED** [review]
Looks good
